exla/lib/exla/op.ex

```diff
@@ -236,6 +236,187 @@ defmodule EXLA.Op do
     %Op{builder: builder, ref: ref}
   end
 
+  @doc """
+  The XLA gather operation stitches together several slices
+  of an input array.
+
+  Note that this operation is extremely generic and far from
+  intuitive for regular usage. However, it can be used to implement
+  many specific operations that have to do with combining multiple
+  tensor slices.
+
+  ## Parameteres
+
+  The XLA docs are rather cryptic unless already understood,
+  so here's an attempt of a more intuitive description.
```

Comment:
Reading the docs for the first time gave me pretty much no idea how this operation works, so instead I looked at the usage across tensorflow and jax codebases and tried various combinations myself until I more or less figured out every parameter. After that the docs made more sense, though still not intuitive. Here I tried my best to describe how each parameter works based on examples, hopefully it helps if someone else wants implement another operation on top of gather.